src/aap_eda/api/serializers/activation.py

```
@@ -1488,3 +1564,46 @@ def _check_injectors(
+    # Check if user provided a rule_engine_credential_id
+    rule_engine_credential_id = data.get("rule_engine_credential_id")
+    if rule_engine_credential_id:
+        # Verify the credential exists
+        try:
+            models.EdaCredential.objects.get(id=rule_engine_credential_id)
+            return
+        except models.EdaCredential.DoesNotExist:
+            raise serializers.ValidationError(
+                f"Rule Engine Credential with id "
+                f"{rule_engine_credential_id} does not exist"
+            )
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)